Tests/PubNubTests/Mocking/MockURLSession.swift

```diff
@@ -59,40 +59,187 @@ class MockURLSessionTask: URLSessionDataTask {
 
   override func resume() {
     mockState = .running
-    (mockSession as? MockURLSession)?.resume(task: self)
+    mockSession?.resume(task: self)
   }
 
   override func cancel() {}
 
-  init(identifier: Int, session: URLSessionReplaceable, request: URLRequest) {
+  init(identifier: Int, session: MockURLSession?, request: URLRequest) {
     mockIdentifier = identifier
     mockSession = session
     mockRequest = request
   }
 }
 
+class MockURLSessionUploadTask: URLSessionUploadTask {
+  weak var mockSession: MockURLSession?
+  var mockIdentifier: Int
+  var mockState: URLSessionTask.State = .suspended
+
+  var mockRequest: URLRequest?
+  var mockResponse: HTTPURLResponse?
+
+  var mockData: Data?
+  var mockError: Error?
+
+  override var response: URLResponse? {
+    return mockResponse
+  }
+
+  override var originalRequest: URLRequest? {
+    return mockRequest
+  }
+
+  override var currentRequest: URLRequest? {
+    return mockRequest
+  }
+
+  override var error: Error? {
+    return mockError
+  }
+
+  override var state: URLSessionTask.State {
+    return mockState
+  }
+
+  override var taskIdentifier: Int {
+    return mockIdentifier
+  }
+
+  override var countOfBytesExpectedToReceive: Int64 {
+    return 0
+  }
+  
```

Comment:
Codacy found an issue: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/PubNub/swift/pullRequest?prid=6302911)